server/src/uuid/model/entity/messages.rs

```diff
@@ -33,6 +37,68 @@ impl MessageResponder for EntityMessage {
     }
 }
 
+#[derive(Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct EntityAddRevisionMutation {
+    pub input: EntityAddRevisionInput,
+    pub revision_type: EntityRevisionType,
+    pub user_id: i32,
+}
+
+#[async_trait]
+impl MessageResponder for EntityAddRevisionMutation {
+    #[allow(clippy::async_yields_async)]
+    async fn handle(&self, connection: Connection<'_, '_>) -> HttpResponse {
+        let payload = EntityAddRevisionPayload {
+            input: EntityAddRevisionInput {
+                changes: self.input.changes.clone(),
+                entity_id: self.input.entity_id,
+                needs_review: self.input.needs_review,
+                subscribe_this: self.input.subscribe_this,
+                subscribe_this_by_email: self.input.subscribe_this_by_email,
+                fields: self.input.fields.clone(),
+            },
+            revision_type: self.revision_type,
+            user_id: self.user_id,
+        };
+
+        let entity_revision = match connection {
+            Connection::Pool(pool) => Entity::add_revision(payload, pool).await,
+            Connection::Transaction(transaction) => {
+                Entity::add_revision(payload, transaction).await
+            }
+        };
+
+        match entity_revision {
+            Ok(_) => HttpResponse::Ok()
+                .content_type("application/json; charset=utf-8")
+                .json(EntityRevisionData {
+                    success: true,
+                    reason: None,
```

Comment:
@inyono What do you think of returning the `id` of the revision? Could that be somehow useful? At any rate, would make testing a little bit easier

Reply:
Good. I'll make it in another PR then.